src/__tests__/host-text-nesting.test.tsx

```diff
@@ -0,0 +1,185 @@
+import * as React from 'react';
+import { Text, Pressable, View } from 'react-native';
+import { render, within } from '../pure';
+
+/**
+ * Our queries interact differently with composite and host elements, and some specific cases require us
+ * to crawl up the tree to a Text composite element to be able to traverse it down again. Going up the tree
+ * is a dangerous behaviour because we could take the risk of then traversing a sibling node to the original one.
+ * This test suite is designed to be able to test as many different combinations, as a safety net.
+ * Specific cases should still be tested within the relevant file (for instance an edge case with `within` should have
+ * an explicit test in the within test suite)
+ */
+describe('nested text handling', () => {
```

Comment:
I see couple of points similar across tests in this files:
1. All `getByRole` tests seem to be more relevant as a check for `*ByRole` queries then text handling, Consider moving them to `role.test.tsx`
2. Ditto `*ByText` queries look like a good candidate to be moved to `text.test.tsx`
3. Using assertions like `expect(view.getByText('Save').props.testID).toBe('text');` seem to be a weaker version of using `expect(view.getByRole('button', { name: 'Save' }).props.testID).toBe(getByTestId('pressable'));` assertion, as the latter would pinpoint exactly to host element. We could use both for the `getByRole` queries as they too return host components

Reply:
My take on this is most of those behaviours are already tested in the different test files (`text.test`, `role.test.tsx`, etc.), but usually without the extra parent/child intermediate components.

So this file is more about adding an additional safety nest, is we care about grouping things, this file doesn't make sense anymore. I'm fine with this, but then we would probably want to create a more exhaustive list of all the test cases we want, see what is missing for the different queries, and adding tests for host vs composite elements a bit everywhere.

I feel this would make our test suites a bit hard to grasp, and rough for contributors to join and contribute (most new features shouldn't have to care about host vs composite).